pkg/blockbuilder/tsdb.go

```diff
@@ -183,6 +192,45 @@ func (b *TSDBBuilder) Process(ctx context.Context, rec *kgo.Record, lastBlockMax
 	return allSamplesProcessed, app.Commit()
 }
 
+// checkTSDBAppendError checks if err is a non-terminal error, that should not block processing other series in the batch.
+func checkTSDBAppendError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	// Check if error is one of the "soft errors" we can proceed on without terminating.
+	// Same as https://github.com/grafana/mimir/blob/1eb4b8e1e3293df100d7fc4df0c94712c31a0930/pkg/ingester/ingester.go#L1283-L1284
```

Comment:
do you want to keep those two functions in sync? did you consider adding a test or moving them to a shared package? Not super likely we change them, but I can see how _if_ we fix one, we're likely to fix the other.

Reply:
Right now almost all of `tsdb.go` is a stripped version of the code from the ingester's tsdb wrapper. We should definitely find a way to share pieces of ingesters, ingest-store and block-builder after it all gets into shape.